### PR TITLE
Docs improvement

### PR DESCRIPTION
**Pull request by ThomSerg**
An attempt to fix some of the issues I found with the documentation
+ inclusion of missing documentation
+ some small additions (like additional cross-reference links)

### CODE REVIEW

**Comment by Wout4** [comment]
Nice, could already be merged or first we do those todo's

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,6 @@
 numpy>=1.17.0
 m2r2>=0.2.7
 sphinx-automodapi
-sphinx_rtd_theme
-sphinx==5.3.0
-sphinx_rtd_theme==1.1.1
-readthedocs-sphinx-search==0.1.1
\ No newline at end of file
+sphinx>=5.3.0 # cannot be updated due to m2r2 no longer being maintained -> switch to MyST?
```
Comment (tias) [inline docs/requirements.txt:4]: I know neither of the two, happy to switch to a maintained theme so we can keep sphinx up to date too yes!

**Review by tias — APPROVED** [review]
super classy, rigorous and with non-trivial catches and additional explanations.

Even includes a todo : ) Maybe go over it with Wout?

Really great. Should be merged asap and definitely before the next release.

**Comment by tias** [comment]
ah, Wout already reviewed it.

Do merge for users to immediately benefit, sphinx theme switch and additional todo's can go in future pull requests.

**Pull request merged**